lms/djangoapps/teams/views.py

```diff
@@ -1117,6 +1118,9 @@ def post(self, request):
                 status=status.HTTP_400_BAD_REQUEST
             )
 
+        if course_module.teams_configuration.get("teams_locked") and not request.user.is_staff:
+            return Response(status=status.HTTP_404_NOT_FOUND)
```

Comment:
This is not super decisive, just asking myself if it's better to return 403

Reply:
for me it's not better because I want to keep the kind of response API handles and when a user has not access it returns 404